osu.Game/Beatmaps/IWorkingBeatmap.cs

```diff
@@ -57,9 +58,10 @@ public interface IWorkingBeatmap
         /// <param name="ruleset">The <see cref="RulesetInfo"/> to create a playable <see cref="IBeatmap"/> for.</param>
         /// <param name="mods">The <see cref="Mod"/>s to apply to the <see cref="IBeatmap"/>.</param>
         /// <param name="timeout">The maximum length in milliseconds to wait for load to complete. Defaults to 10,000ms.</param>
+        /// <param name="timeoutToken">Cancellation token that cancels the beatmap loading process.</param>
         /// <returns>The converted <see cref="IBeatmap"/>.</returns>
         /// <exception cref="BeatmapInvalidForRulesetException">If <see cref="Beatmap"/> could not be converted to <paramref name="ruleset"/>.</exception>
-        IBeatmap GetPlayableBeatmap(RulesetInfo ruleset, IReadOnlyList<Mod> mods = null, TimeSpan? timeout = null);
+        IBeatmap GetPlayableBeatmap(RulesetInfo ruleset, IReadOnlyList<Mod> mods = null, TimeSpan? timeout = null, CancellationToken timeoutToken = default);
```

Comment:
Any reason this is named `timeoutToken` here?

Also we can't probably remove/replace the `timeout` parameter as part of this change, may feel weird with both.

Reply:
I've [suggested](https://github.com/ppy/osu/pull/15494#discussion_r744040546) that the timeout token created inside *should be* renamed `timeoutToken`, but the one passed to the method itself should remain named `cancellationToken` as it was.

Reply:
See the rest of my review. The inner one should likely be removed.

Reply:
@Tollii why was this conversation resolved? the comments above mention that one of the tokens should be removed but i don't see that happening here, or an explanation why that shouldn't be done.

the fact that the timeout is silently defaulting to 10 secs via `createCancellationTokenSource()` may be a valid reason not to remove the inner, but i'd prefer that that was stated explicitly rather than left to guesswork.